sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -648,6 +725,16 @@ pi_result piDeviceRelease(pi_device Device) {
   if (--(Device->RefCount) == 0) {
     // Destroy the command list used for initializations
     ZE_CALL(zeCommandListDestroy(Device->ZeCommandListInit));
+    // Destroy all the command lists associated with this device.
+    Device->ZeCommandListCacheMutex.lock();
```

Comment:
Because I want to hold the lock for the minimum amount of time, I cannot rely on the smart pointer destruction to unlock for me. I would like to keep the explicit lock/unlock indications for deterministic locking sequencing.

Reply:
You can use scopes (aka {}) to limit life of objects.

Reply:
Because of the different scopes for the locks such as in getAvailableCommandList I would prefer that we keep the explicit lock/unlock unless we rework that code. Please let me know if this is a blocker & thank you for your time and comments!

Reply:
It is not a blocker.